src/Services/PaymentServiceInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Gcd\Scaffold\Payments\Services;
+
+interface PaymentServiceInterface
+{
+    public function startPayment($APIKey, PaymentServiceEntity $entity);
```

Comment:
I think $APIKey might be too simple - some providers may have multiple credentials or setups (e.g. Realex needs a secret and a merchant ID). On my side I was using a StripeSettings singleton to track to provider configuration. Would that suffice?